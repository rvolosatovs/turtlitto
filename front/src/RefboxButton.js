import React, { Component } from 'react';
import styled from "styled-components";

/**
 * A button part of the refbox. Does a callback to its parent to handle onClick events
 * @param {*} props Contains the background color of the button, and the text the button needs to display
 * Author: S.A. Tanja
 */
const RefboxButton = (props) => {
    return (
        <Button
            color = {props.color}
            onClick = {() => {props.onClick(props.color, props.children)}}
        >{props.children}</Button>
    );
};

/**
 * A styled component defining the style of the refbox buttons
 */
const Button = styled.button`
    background-color : ${props => props.color};
    color : black;
    width : 206px;
    height : 206px;
    border : solid;
    border-width : 4px;
    text-align : center;
    font-size : 100px;
`;



export default RefboxButton;