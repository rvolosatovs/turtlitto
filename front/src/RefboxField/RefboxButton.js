import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

/**
 * A button part of the refbox. Does a callback to its parent to handle onClick events
 * Author: S.A. Tanja
 * Author: G.W. van der Heijden
 *
 * Props:
 *  - teamColor: the team color of the button (cyan or magenta).
 *  - onClick: a function on what to do when the button is pressed
 *  - tag: the tag of the button according to the TRC.
 */
const RefboxButton = props => {
  return (
    <Button
      teamColor={props.teamColor}
      onClick={() => {
        props.onClick();
      }}
    >
      {props.tag}
    </Button>
  );
};

/**
 * A styled component defining the style of the refbox buttons
 */
const Button = styled.button`
  background-color: ${props => props.teamColor};
  width: 20.6rem;
  height: 20.6rem;
  border: 0.25rem solid;
  font-size: 10rem;
`;

RefboxButton.propType = {
  teamColor: PropTypes.oneOf(["cyan", "magenta"]).isRequired,
  tag: PropTypes.oneOf(["KO", "FK", "GK", "TI", "C", "P"]).isRequired
};

export default RefboxButton;
