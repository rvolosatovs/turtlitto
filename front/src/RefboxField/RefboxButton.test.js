import React from "react";
import RefboxButton from "./RefboxButton";
import renderer from "react-test-renderer";

describe("RefboxButton", () => {
  it("shows a cyan button with KO on it", () => {
    const component = renderer.create(
      <RefboxButton teamColor="cyan" onClick={() => {}} tag="KO" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("shows a magenta button with P on it", () => {
    const component = renderer.create(
      <RefboxButton teamColor="magenta" onClick={() => {}} tag="P" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
