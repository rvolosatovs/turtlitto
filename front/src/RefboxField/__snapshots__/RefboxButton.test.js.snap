// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefboxButton shows a cyan button with KO on it 1`] = `
<button
  className="sc-bdVaJa jBNwEU"
  onClick={[Function]}
>
  KO
</button>
`;

exports[`RefboxButton shows a magenta button with P on it 1`] = `
<button
  className="sc-bdVaJa iwQbdo"
  onClick={[Function]}
>
  P
</button>
`;
